### DISCUSSION TITLE
How to use generate_java_gadget

### DISCUSSION THREAD

**Discussion by j4vaovo** [projectdiscovery/nuclei#3614] [Q&A]
<img width="614" alt="aaa" src="https://user-images.githubusercontent.com/128683738/234469612-d9833173-369c-4706-97c9-109abbf21da4.png">
<img width="635" alt="bbb" src="https://user-images.githubusercontent.com/128683738/234469618-1e27dc43-ad36-41de-a9b5-630d3161060a.png">
<img width="470" alt="ddd" src="https://user-images.githubusercontent.com/128683738/234469627-ca861224-ca39-4604-988a-285dee6ad853.png">
<img width="777" alt="ccc" src="https://user-images.githubusercontent.com/128683738/234469641-763d287c-3f59-4d41-818b-87b7e8e4ab4f.png">

why didn t my modification work

**Comment by j4vaovo** [comment]
@tarunKoyalwar

**Comment by ehsandeep** [comment]
@j4vaovo, you need to specify any of the supported payload types, instead you have used "aaaa".

Here is list of supported payload - https://nuclei.projectdiscovery.io/templating-guide/helper-functions/#deserialization-helper-functions

- dns
- commons-collections3.1
- commons-collections4.0
- jdk7u21
- jdk8u20
- groovy1

**Comment by j4vaovo** [comment]
@ehsandeep 
<img width="462" alt="搜狗高速浏览器截图20230426202320" src="https://user-images.githubusercontent.com/128683738/234573445-8ad2e948-27a2-4576-869b-6e206160b3f6.png">
I changed java.go and recompiled it but it didn't work
I'm a golang rookie

**Comment by j4vaovo** [comment]
https://github.com/projectdiscovery/nuclei/issues/3615

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
